### PR TITLE
starting on an empty requirements file

### PR DESCRIPTION
**Pull request by peterbe**

### CODE REVIEW

File: tests.py [hunk 1]
```diff
@@ -176,3 +205,66 @@ def test_amend_requirements_content_replacement_amonst_others_2(self):
...
         self.assertEqual(result, previous + new_lines)
+
+    @httpretty.activate
+    @cleanup_tmpdir('peepin*')
+    def test_run(self):
+        httpretty.register_uri(
+            httpretty.GET,
+            "https://pypi.python.org/pypi/peepin",
+            body="""
+            <div id="content">
```
Comment (erikrose) [inline tests.py]: Did you want to close this eventually?

File: tests.py [hunk 2]
```diff
@@ -176,3 +205,66 @@ def test_amend_requirements_content_replacement_amonst_others_2(self):
...
+            "https://pypi.python.org/packages/3.3/p/peepin/peepin-0.10-py3-none-any.whl#md5=45",
+            body="Some py3 wheel content\n",
+        )
+        httpretty.register_uri(
+            httpretty.GET,
+            "https://pypi.python.org/packages/source/p/peepin/peepin-0.10.tar.gz#md5=ae",
+            body="Some tarball content\n",
+        )
+        with tmpfile() as filename:
+            open(filename, 'w').write('')
```
Comment (erikrose) [inline tests.py]: Why not use `with open` here?

File: tests.py [hunk 3]
```diff
@@ -176,3 +205,66 @@ def test_amend_requirements_content_replacement_amonst_others_2(self):
...
+            retcode = peepin.run('peepin==0.10', filename)
+            self.assertEqual(retcode, 0)
+            output = open(filename).read()
```
Comment (erikrose) [inline tests.py]: Not strictly necessary, but it would be good hygiene to close this file handle as well.

File: tests.py [hunk 4]
```diff
@@ -176,3 +205,66 @@ def test_amend_requirements_content_replacement_amonst_others_2(self):
...
+<a href="https://pypi.python.org/packages/3.3/p/peepin/peepin-0.10-py3-none-any.whl#md5=45"
+    >peepin-0.10-py3-none-any.whl</a>
+
+<a href="https://pypi.python.org/packages/source/p/peepin/peepin-0.10.tar.gz#md5=ae"
+    >peepin-0.10.tar.gz</a>
+            """,
+        )
+        httpretty.register_uri(
+            httpretty.GET,
+            "https://pypi.python.org/packages/2.7/p/peepin/peepin-0.10-py2-none-any.whl#md5=0a",
```
Comment (erikrose) [inline tests.py]: Those are really short MD5 hashes. What's up with that?

**Comment by erikrose** [comment]
Otherwise, looks sane to me!

File: tests.py (hunk 1 above)
Comment (peterbe) [inline tests.py]: Doesn't matter. It's a regex that looks for [certain hrefs](https://github.com/peterbe/peepin/blob/master/peepin.py#L117)

File: tests.py (hunk 2 above)
Comment (peterbe) [inline tests.py]: Does that matter in test code? After all, file descriptors will be cleaned up after finishing the tests. 
I thought properly closing is only relevant in daemon-like code or code with infinite loops like tornado.

File: tests.py (hunk 3 above)
Comment (peterbe) [inline tests.py]: It does close the file as soon as the test ends.

File: tests.py (hunk 4 above)
Comment (peterbe) [inline tests.py]: I truncated them so the lines wouldn't be so darn long.

**Comment by peterbe** [comment]
Grr... I can't get the tests to ever finish for py 3.4. I wonder if it's a coincidence that it's first in the list. It always seems to hang for 10 minutes after it concludes that all tests passed fine.

**Pull request merged**